test/conftest.py

```diff
@@ -59,3 +61,17 @@ def cache_dir():
     cache = tempfile.TemporaryDirectory()
     yield cache.name
     cache.cleanup()
+
+
+@pytest.fixture
+def req_file():
+    req_file = tempfile.NamedTemporaryFile(delete=False)
+    yield Path(req_file.name)
+    os.remove(req_file.name)
+
+
+@pytest.fixture
+def other_req_file():
+    req_file = tempfile.NamedTemporaryFile(delete=False)
+    yield Path(req_file.name)
+    os.remove(req_file.name)
```

Comment:
N.B.: You can avoid two fixtures here by making this into a higher-order fixture, e.g.:

```python
@pytest.fixture
def req_file():
    def _req_file():
        # original body here
    return _req_file
```

I'll do that as part of merge fixups.